### PR TITLE
bump NFD_Operato GO versions

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by jmencak** [comment]
Thank you for the PR, Ed.  Do we also need to do the same change for `openshift-cluster-nfd-operator-release-4.8.yaml` and possibly `openshift-cluster-nfd-operator-release-4.9.yaml`? Letting Kevin comment. :)

**Comment by kpouget** [comment]
> Thank you for the PR, Ed. Do we also need to do the same change for `openshift-cluster-nfd-operator-release-4.8.yaml` and possibly `openshift-cluster-nfd-operator-release-4.9.yaml`? Letting Kevin comment. :)

>  Letting Kevin comment. :)

==> I have no clue :# but it might be the opportunity to understand more about it!
this operator is built by ART I understand, do you have any pointer to the Dockerfile it uses / where it's being built?

but most likely the answer will be yes, I think `4.8` and `4.9` should be updated as well, no reason they stay with an old version of Go :)

**Comment by ArangoGutierrez** [comment]
@jmencak @kpouget done :)

**Comment by ArangoGutierrez** [comment]
@kpouget @jmencak : 
This PR targets `image` test which is running to completion (see [ci/images)](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/17585/rehearse-17585-pull-ci-openshift-cluster-nfd-operator-master-images/1382797823040294912) 
I think we can move forward, I have checked with the CI team and they confirm this is just a flake causing aws and AWS-upgrade to fail , tests that are not being modified on this PR.

cc @alvaroaleman @stevekuznetsov

**Comment by ArangoGutierrez** [comment]
I am 100% confident that this  patch works, I have my PC running GO 1.16.3, and have been building 1.16 images for 2 weeks now (both upstream and downstream

Upstream https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/n[…]covery-operator/node-feature-discovery-operator-presubmits.yaml

has been running on 1.16 for weeks now

**Comment by kpouget** [comment]
> I am 100% confident that this patch works, I have my PC running GO 1.16.3, and have been building 1.16 images for 2 weeks now (both upstream and downstream
> 
> Upstream [https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/n[…]covery-operator/node-feature-discovery-operator-presubmits.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/n%5B%E2%80%A6%5Dcovery-operator/node-feature-discovery-operator-presubmits.yaml)
> 
> has been running on 1.16 for weeks now

thanks @ArangoGutierrez , I think you know enough about the repo to get this one merged!

\lgtm

**Comment by jmencak** [comment]
fixing the Windows slash :)
/lgtm

**Comment by jmencak** [comment]
/lgtm cancel
Sorry, I was too quick, this doesn't look right. Are you sure about `registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8`

**Comment by jmencak** [comment]
Also, how do we check `tag: golang-1.16` exists?

**Comment by ArangoGutierrez** [comment]
> /lgtm cancel
> Sorry, I was too quick, this doesn't look right. Are you sure about `registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8`

it does exsists :
https://github.com/openshift/cluster-nfd-operator/blob/master/Dockerfile#L2
if you have your podman logged as in https://gist.github.com/ArangoGutierrez/42f393ddae682707892125d228824416 `podman build` works

**Comment by kpouget** [comment]
@ArangoGutierrez @jmencak 
this page https://prow.ci.openshift.org/pr-history/?org=openshift&repo=release&pr=17585 gives an overview of the PR history,
and actually, all the tests passed at some point (with different commits though) (and `rehearse` doesn't count as it's green only if all the test pass at once I think)

File: ci-operator/config/openshift/cluster-nfd-operator/openshift-cluster-nfd-operator-release-4.9.yaml [hunk 1]
```diff
@@ -1,37 +1,21 @@
 base_images:
   base:
-    name: "4.9"
+    name: "4.8"
```
Comment (jmencak) [inline ci-operator/config/openshift/cluster-nfd-operator/openshift-cluster-nfd-operator-release-4.9.yaml:3]: Why the change from 4.9->4.8?

Comment (ArangoGutierrez) [reply ci-operator/config/openshift/cluster-nfd-operator/openshift-cluster-nfd-operator-release-4.9.yaml:3]: base 4.9 doesn't exists yet

**Comment by jmencak** [comment]
> > /lgtm cancel
> > Sorry, I was too quick, this doesn't look right. Are you sure about `registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.8`
> 
> it does exsists :
> https://github.com/openshift/cluster-nfd-operator/blob/master/Dockerfile#L2
> if you have your podman logged as in https://gist.github.com/ArangoGutierrez/42f393ddae682707892125d228824416 `podman build` works

Sure it does, because you're using `registry.ci.openshift.org` and not `registry.svc.ci.openshift.org` there.

**Comment by ArangoGutierrez** [comment]
You were right @jmencak

**Comment by jmencak** [comment]
This cleanup makes sense to me. Thank you for the PR, Ed.
/lgtm
/hold
Feel free to cancel the hold just in case you don't want to squash your commits.
